Failing test

Even in python 2.7, the one test fails because of the following line:

https://github.com/arfc/annsa/blob/b3b7eca913b72ec584c2deb7c7e16db3b72095db/annsa/tests/test_annsa.py#L8

```
data_path = op.join(sb.__path__[0], 'data')
```

sb doesn't exist. What's the desired behavior here?


The current behavior is:

```

(py27) khuff@maria:~/repos/annsa(ci) $ py.test
============================================================================ test session starts =============================================================================
platform darwin -- Python 2.7.11, pytest-2.8.1, py-1.4.30, pluggy-0.3.1
rootdir: /Users/khuff/repos/annsa, inifile:
collected 0 items / 1 errors

=================================================================================== ERRORS ===================================================================================
_________________________________________________________________ ERROR collecting annsa/tests/test_annsa.py _________________________________________________________________
annsa/tests/test_annsa.py:8: in <module>
    data_path = op.join(sb.__path__[0], 'data')
E   NameError: name 'sb' is not defined
========================================================================== 1 error in 3.22 seconds ===========================================================================
```

--- a/.circleci/config.yml
+++ b/.circleci/config.yml
@@ -8,7 +8,7 @@ jobs:
     docker:
       # specify the version you desire here
       # use `-browsers` prefix for selenium tests, e.g. `3.6.1-browsers`
-      - image: circleci/python:3.6.1
+      - image: circleci/python:2.7
       
       # Specify service dependencies here if necessary
       # CircleCI maintains a library of pre-built images

--- a/.circleci/config.yml
+++ b/.circleci/config.yml
@@ -1,0 +1,56 @@
+# Python CircleCI 2.0 configuration file
+#
+# Check https://circleci.com/docs/2.0/language-python/ for more details
+#
+version: 2
+jobs:
+  build:
+    docker:
+      # specify the version you desire here
+      # use `-browsers` prefix for selenium tests, e.g. `3.6.1-browsers`
+      - image: circleci/python:3.6.1
+      
+      # Specify service dependencies here if necessary
+      # CircleCI maintains a library of pre-built images
+      # documented at https://circleci.com/docs/2.0/circleci-images/
+      # - image: circleci/postgres:9.4
+
+    working_directory: ~/annsa
+
+    steps:
+      - checkout
+
+      # Download and cache dependencies
+      - restore_cache:
+          keys:
+          - v1-dependencies-{{ checksum "requirements.txt" }}
+          # fallback to using the latest cache if no exact match is found
+          - v1-dependencies-
+
+      - run:
+          name: install dependencies
+          command: |
+            python3 -m venv venv
+            . venv/bin/activate
+            pip install -r requirements.txt
+
+      - save_cache:
+          paths:
+            - ./venv
+          key: v1-dependencies-{{ checksum "requirements.txt" }}
+        
+      # run tests!
+      # this example uses Django's built-in test-runner
+      # other common Python testing frameworks include pytest and nose
+      # https://pytest.org
+      # https://nose.readthedocs.io
+      - run:
+          name: run tests
+          command: |
+            . venv/bin/activate
+            py.text
+
+      - store_artifacts:
+          path: test-reports
+          destination: test-reports
+          

--- a/annsa/tests/test_annsa.py
+++ b/annsa/tests/test_annsa.py
@@ -5,14 +5,15 @@ import pandas as pd
 import numpy.testing as npt
 import annsa as an
 
+
 def test_generate_single_source_key():
     """
-    Testing to see if the generate_single_source_key function doesn't give infinite or negative values.
+    Testing to see if the generate_single_source_key function doesn't give
+    infinite or negative values.
 
     """
-    # Make a blank source key 
+    # Make a blank source key
     blank_source_key = an.generate_single_source_key()
     # Check if key is blank
     inf_check = np.any(np.isinf(blank_source_key))
     npt.assert_equal(inf_check, False)
-    

--- a/annsa/tests/test_annsa.py
+++ b/annsa/tests/test_annsa.py
@@ -5,9 +5,6 @@ import pandas as pd
 import numpy.testing as npt
 import annsa as an
 
-data_path = op.join(sb.__path__[0], 'data')
-
-
 def test_generate_single_source_key():
     """
     Testing to see if the generate_single_source_key function doesn't give infinite or negative values.

--- a/requirements.txt
+++ b/requirements.txt
@@ -2,3 +2,4 @@ numpy
 scipy
 matplotlib
 pandas
+pytest

--- a/requirements.txt
+++ b/requirements.txt
@@ -3,3 +3,6 @@ scipy
 matplotlib
 pandas
 pytest
+tensorflow
+sklearn
+pandas

